packages/core/src/internal/execution/future-processor/helpers/build-initialize-message-for.ts

```diff
@@ -226,18 +234,21 @@ function _extendBaseInitWith<
 >(
   messageType: MessageT,
   future: FutureT,
-  strategy: { name: string },
+  strategy: string,
+  strategyConfig: Record<PropertyKey, string | number>,
```

Comment:
This is an internal type. It covers the current possibilities, and can be expanded to recursive types with the Strategy API.